Season 03/Jungmin/Week 24/week24_2156.java

```diff
@@ -0,0 +1,29 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class week24_2156 {
+
+    public static int n;
+    public static int[] wine, dpArr;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        n = Integer.parseInt(br.readLine());
+
+        wine = new int[10001];
+        dpArr = new int[10001];
+
+        for (int i = 1; i <= n; i++) wine[i] = Integer.parseInt(br.readLine());
+
+        dpArr[1] = wine[1];
+        dpArr[2] = wine[1] + wine[2];
+
+        for (int i = 3; i <= n; i++) {
+            // i번째 와인을 마시는 것이 최솟값이 아닐 수도 있음 -> i-1번째와 비교
+            dpArr[i] = Math.max(Math.max(dpArr[i-3] + wine[i-1], dpArr[i-2]) + wine[i], dpArr[i-1]);
+        }
+
```

Comment:
```suggestion
        int now = 0;
        int before = 0;
        for (int i = 1; i <= n; i++) {
           now = Integer.parseInt(br.readLine());
           if ( i == 1 ) dpArr[1] = now;
           else if( i == 2 ) dpArr[2] = before + now;
           else {
               dpArr[i] = Math.max(Math.max(dpArr[i-3] + before, dpArr[i-2]) + now, dpArr[i-1]);           
           }
           before = now;
        }

```

요렇게 한번에 읽으면서 계산도 가능하겠네요 !  !  !

Reply:
대박 그러네요 !! now와 before라니 생각도 못함 이마 빡빢빡